client/client-multi/client-c/src/kaa_event.c

```diff
@@ -732,6 +905,38 @@ kaa_error_t kaa_event_handle_server_sync(kaa_event_manager_t *self
     return KAA_ERR_NONE;
 }
 
+kaa_error_t kaa_event_manager_find_event_listeners(kaa_event_manager_t *self, const char *fqns[], size_t fqns_count, const kaa_event_listeners_callback_t *callback)
+{
+    KAA_RETURN_IF_NIL3(self, fqns_count, callback, KAA_ERR_BADPARAM);
+    KAA_RETURN_IF_NIL2(callback->on_event_listeners, callback->on_event_listeners_failed, KAA_ERR_BADPARAM);
+
+    kaa_event_listeners_request_t *subscriber = create_event_listener_request(++self->event_listeners_request_id
+                                                                               , fqns, fqns_count
+                                                                               , callback);
+    KAA_RETURN_IF_NIL(subscriber, KAA_ERR_NOMEM);
+
+    if (!self->event_listeners_requests) {
```

Comment:
kaa_list_t *request_it = self->event_listeners_requests ? 
kaa_list_push_front(self->event_listeners_requests, subscriber) : 
kaa_list_create(subscriber);

if (!request_it) {
    destroy_event_listener_request(subscriber);
    --self->event_listeners_request_id;
    return KAA_ERR_NOMEM;
}

self->event_listeners_requests = it;